src/main/java/net/neoforged/neoforge/common/util/NeoForgeExtraCodecs.java

```diff
@@ -30,8 +33,10 @@
  * @see ExtraCodecs
  */
 public class NeoForgeExtraCodecs {
-    public static final Codec<Optional<WithConditions<Recipe<?>>>> CONDITIONAL_RECIPE_CODEC = ConditionalOps.createConditionalCodecWithConditions(Recipe.CODEC);
-    public static final Codec<Optional<WithConditions<Advancement>>> CONDITIONAL_ADVANCEMENT_CODEC = ConditionalOps.createConditionalCodecWithConditions(Advancement.CODEC);
+    //Lazily initialize these codecs so that loading NeoForgeExtraCodecs in things like JUnit don't cause cascading class loading of things that
+    // can't be initialized properly such as registries
+    public static final Codec<Optional<WithConditions<Recipe<?>>>> CONDITIONAL_RECIPE_CODEC = ExtraCodecs.lazyInitializedCodec(() -> ConditionalOps.createConditionalCodecWithConditions(Recipe.CODEC));
+    public static final Codec<Optional<WithConditions<Advancement>>> CONDITIONAL_ADVANCEMENT_CODEC = ExtraCodecs.lazyInitializedCodec(() -> ConditionalOps.createConditionalCodecWithConditions(Advancement.CODEC));
```

Comment:
JUnit tests should ideally call `Bootstrap.init()` and also start mod loading. With official JUnit coming soon:tm:, I don't think this should be changed.

Reply:
These codecs should not be in `NeoForgeExtraCodecs` in the first place, IMO.

Reply:
Agreed, I think moving them to the relevant classes is reasonable so I will go ahead and do so.